sdk/core/Azure.Core/tests/NextLinkOperationImplementationTests.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core.Pipeline;
+using NUnit.Framework;
+
+namespace Azure.Core.Tests
+{
+    public class NextLinkOperationImplementationTests
+    {
+        [Test]
+        public void ConstructRehydrationTokenTest()
+        {
+            var requetMethod = RequestMethod.Get;
+            var startRequestUri = new Uri("https://test");
+            var nextRequestUri = "nextRequestUri";
+            var headerSource = "None";
+            string lastKnownLocation = null;
+            var finalStateVia = OperationFinalStateVia.OperationLocation.ToString();
+            var token = NextLinkOperationImplementation.GetRehydrationToken(requetMethod, startRequestUri, nextRequestUri, headerSource, lastKnownLocation, finalStateVia);
+            Assert.AreEqual(requetMethod, token.RequestMethod);
+            Assert.AreEqual(startRequestUri, token.InitialUri);
+            Assert.AreEqual(nextRequestUri, token.NextRequestUri);
+            Assert.AreEqual(headerSource, token.HeaderSource);
+            Assert.AreEqual(lastKnownLocation, token.LastKnownLocation);
+            Assert.AreEqual(finalStateVia, token.FinalStateVia);
+        }
+
+        [Test]
+        public void ConstructNextLinkOperationTest()
+        {
+            var rehydrationToken = new RehydrationToken(null, null, "None", "nextRequestUri", "https://test", RequestMethod.Delete, null, OperationFinalStateVia.AzureAsyncOperation.ToString());
+            var operation = NextLinkOperationImplementation.Create(HttpPipelineBuilder.Build(new MockClientOptions()), rehydrationToken, null);
+            Assert.NotNull(operation);
```

Comment:
I wish there was more that we could validate here to ensure that it was created correctly, but I don't see a way to do so.

@m-nash:  Any thoughts?